middleware/index.ts

```diff
@@ -102,27 +103,62 @@ async function invokeAsyncFn(logger: Logger, payload: any, headers: any): Promis
             req.destroy(expectedErr);
         });
     });
-};
+}
+
+/**
+ * Take all input headers and convert them to map of lower-case key
+ * to input string value.
+ *
+ * @param message riff Message
+ */
+function toLowerCaseKeyHeaders(message: any): ReadonlyMap<string,string> {
+    const hdrs = message['headers'];
+    const hmap = new Map<string,string>();
+    Object.keys(hdrs.toRiffHeaders()).forEach((key) => {
+        const lcKey = key.toLowerCase();
+        hmap[lcKey] = hdrs.getValue(key);
+    });
+    return hmap;
+}
 
 const userFn = loadUserFunction();
 
 export default async function systemFn(message: any): Promise<any> {
-    const payload = message['payload'];
+    // Remap riff headers to a standard JS object with lower-case keys
+    const headers = toLowerCaseKeyHeaders(message);
 
-    // Remap riff headers to a standard JS object
-    const headers = message['headers'].toRiffHeaders();
-    Object.keys(headers).map((key: string) => {headers[key] = message['headers'].getValue(key)});
+    // evergreen:function:invoke includes an extra 'data' level for BinaryHTTP format
+    let bodyPayload: any = message['payload'];
+    if ('data' in bodyPayload && 'ce-id' in headers && headers['ce-specversion'] === '0.3') {
+        bodyPayload = bodyPayload['data'];
+    }
 
-    const requestId = headers['Ce-Id'] || headers['X-Request-Id'];
+    // Initialize logger with request ID
+    const requestId = headers['ce-id'] || headers['x-request-id'] || bodyPayload['id'];
     const requestLogger = createLogger(requestId);
+  
+    // Parse input according to Cloudevents 0.3 or 1.0 specification
+    let cloudEvent: CloudEvent;
+    try {
+        cloudEvent = httpReceiver.accept(headers, bodyPayload);
```

Comment:
Oh, good find.

Reply:
Updated in most recent push - needs special conditions since cloudevents-sdk 2.x does *not* handle `0.2` format events sent by core 228.